modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/RuntimeMessage.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2017-2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.runtime;
+
+/**
+ * Represents a message to be held by the Runtime framework.
+ * <p>
+ * Allows detecting and displaying errors and warnings at server startup and when hot-reloading.
+ *
+ * @since 11.3
+ */
+public class RuntimeMessage {
+
+    protected final Level level;
+
+    protected final String message;
+
+    public RuntimeMessage(Level level, String message) {
+        this.level = level;
+        this.message = message;
+    }
+
+    public Level getLevel() {
+        return level;
+    }
+
+    public String getMessage() {
+        return message;
+    }
+
+    public enum Level {
+
+        ERROR,
+
+        WARNING
+
+    }
+
+    protected enum ComponentManagerStep {
+
+        ACTIVATING,
+
+        STARTING,
+
+        RUNNING,
+
+        STOPPING,
+
+        DEACTIVATING
+
+    }
```

Comment:
Thanks for the `RuntimeMessage#step` cleanup, I didn't see any usage except the `RuntimeMessageHandlerImpl`, does it worth to leave this class over there?

Reply:
Will move it to `RuntimeMessageHandlerImpl` :+1: